src/Types/Transaction.purs

```diff
@@ -74,55 +61,97 @@ newtype Ed25519Signature = Ed25519Signature String -- (bech32)
 
 newtype PlutusScript = PlutusScript String
 
-newtype PlutusData = PlutusData String 
+newtype PlutusData = PlutusData String
 -- TODO - we need a capability to encode/decode Datum from/to serialized format
--- see `makeIsDataIndexed` 
+-- see `makeIsDataIndexed`
 
 newtype Redeemer = Redeemer
   { tag :: RedeemerTag, -- ScriptPurpose: 'spending' 'minting' etc
-    index :: BigInt.BigInt,
+    index :: BigInt,
     data :: PlutusData,
     ex_units :: (MemExUnits /\ CpuExUnits)
   }
 
-newtype MemExUnits = MemExUnits BigInt.BigInt
+newtype MemExUnits = MemExUnits BigInt
 
-newtype CpuExUnits = CpuExUnits BigInt.BigInt
+newtype CpuExUnits = CpuExUnits BigInt
 
 data RedeemerTag = Spend | Mint | Cert | Reward
 
 type AuxiliaryData = Unit -- this is big and weird in serialization-lib
 
-
 newtype TransactionInput = TransactionInput
   { transaction_id :: String, -- TransactionHash
-    index :: BigInt.BigInt -- u32 TransactionIndex
+    index :: BigInt -- u32 TransactionIndex
   }
+derive instance eqTransactionInput :: Eq TransactionInput
+derive instance genericTransactionInput :: Generic TransactionInput _
+derive instance ordTransactionInput :: Ord TransactionInput
+
+instance showTransactionInput :: Show TransactionInput where
+  show = genericShow
 
 newtype TransactionOutput = TransactionOutput
   { address :: Address,
     amount :: Value,
     data_hash :: Maybe String -- DataHash>,
   }
+derive instance eqTransactionOutput :: Eq TransactionOutput
+derive instance genericTransactionOutput :: Generic TransactionOutput _
+derive instance newtypeTransactionOutput :: Newtype TransactionOutput _
+
+instance showTransactionOutput :: Show TransactionOutput where
+  show = genericShow
+
+newtype UtxoM = UtxoM Utxo
+derive instance newtypeUtxoM :: Newtype UtxoM _
+derive newtype instance showUtxoM :: Show UtxoM
 
-newtype Coin = Coin BigInt.BigInt
+type Utxo = Map TransactionInput TransactionOutput
 
-newtype Slot = Slot BigInt.BigInt
+newtype Coin = Coin BigInt
+derive instance eqCoin :: Eq Coin
+derive instance newtypeCoin :: Newtype Coin _
 
-newtype Address = Address 
+newtype Slot = Slot BigInt
+derive instance eqSlot :: Eq Slot
+
+newtype Address = Address
   { "AddrType" :: BaseAddress
   }
+derive instance eqAddress :: Eq Address
```

Comment:
just thinking for all these instances...  shouldn't they be newtype instances?

Reply:
i believe this reduces code size.

Reply:
Makes sense, I will look to change them all.